js/packages/teams-ai/src/models/OpenAIModel.ts

```diff
@@ -449,9 +575,22 @@ export class OpenAIModel implements PromptCompletionModel {
         return params;
     }
 
-    private getInputMessage(messages: Message<string>[]): Message<string> | undefined {
+    private getInputMessage(messages: Message<string>[]): Message<string>[] | Message<string> | undefined {
         const last = messages.length - 1;
-        if (last > 0 && messages[last].role == 'user') {
+        if (last > 0 && messages[last].role !== 'assistant') {
+            // Handling for when there are multiple action output responses (e.g. user message instantiated multiple tool calls)
+            if (messages[last].role === 'tool') {
+                const toolsInput: Message<string>[] = [];
+
+                for (let i = messages.length - 1; i >= 0; i--) {
+                    if (messages[i].action_calls) {
+                        break;
+                    }
+                    toolsInput.push(messages[i]);
```

Comment:
does push add to the end of the array? this will change the order of the tool calls

Reply:
Oh thanks for the catch.